internal/server/server.go

```diff
@@ -30,9 +32,21 @@ func initialDatabase() {
 // Server method handles all requests
 func Server(port string) {
 	engine := html.New("internal/handlers/views", ".html")
+	var errMessage models.Error
 	
 	app := fiber.New(fiber.Config{
 		Views: engine,
+		ErrorHandler: func(ctx *fiber.Ctx, err error) error {
+			if errors.Is(err, sql.ErrNoRows)  {
+				errMessage = models.Error{Message: "Page Not Found"}
+				ctx.Redirect("/404", 500)
```

Comment:
The following instruction (after this one) will not be reached, I don't think. I really dislike re-directs other that for logging in - personal opinion. I think it would be better just to display a basic 404 template. You can set HTTP status using ctx.Status()

Reply:
Ok... Thanks Dima for this